internal/jar/manifest.go

```diff
@@ -0,0 +1,43 @@
+//
+// Copyright 2022.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jar
+
+import "fmt"
+
+const (
+	implementationTitle   = "Implementation-Title"
+	implementationVersion = "Implementation-Version"
+)
+
+type Manifest struct {
+	properties map[string]string
+}
+
+func (m *Manifest) GetImplementationVersion() (string, error) {
+	if val, ok := m.properties[implementationVersion]; !ok {
+		return "", fmt.Errorf("Implementation Version not found in manifest")
+	} else {
+		return val, nil
```

Comment:
Drop the else, just:

```
}

return val, nil
```

Reply:
doesn't `val` need to be accessible from the else?

Reply:
Was thinking something like this, but either is fine honestly. Feel free to ignore.
```
val, ok := ...
if !ok {
 return "", err
}
return val, nil
```

Reply:
oh I see, yeah that's a little nicer I think